src/test/scala/io/iohk/ethereum/mpt/MerklePatriciaTrieSuite.scala

```diff
@@ -594,4 +602,68 @@ class MerklePatriciaTrieSuite extends AnyFunSuite with ScalaCheckPropertyChecks
     assert(trieAtBlock10.get(decodeHexString("bbbb")).contains(largeByteString))
     assert(trieAtBlock10.get(decodeHexString("bbba")).contains(ByteString()))
   }
+
+  test("getProof returns empty result from an empty tree") {
+    val emptyTrie = MerklePatriciaTrie[Int, Int](EmptyEphemNodeStorage)
+    val proof = emptyTrie.getProof(key = 0)
+    assert(proof.isEmpty)
+  }
+
+  test("getProof returns empty result for not existing key") {
+    // given
+    val key1: Array[Byte] = Hex.decode("10000001")
+    val key2: Array[Byte] = Hex.decode("10000002")
+    val key3: Array[Byte] = Hex.decode("10000003")
+
+    val val1: Array[Byte] = Hex.decode("0101")
+    val val2: Array[Byte] = Hex.decode("0102")
+    val val3: Array[Byte] = Hex.decode("0103")
+
+    val trie = EmptyTrie
+      .put(key1, val1)
+      .put(key2, val2)
+      .put(key3, val3)
+
+    // when
+    val proof = trie.getProof(Hex.decode("00000001"))
+
+    // then
+    assert(proof.isEmpty)
+  }
+
+  test("getProof returns valid proof for existing key") {
```

Comment:
It would be nice to have a generator for Merkle Patricia Trees so we could check properties instead of case by case.

Reply:
It is a bit tricky to make MPT generator, because MPT is a persistend data structure - this means it store in some persistend store like DB (in our case rocksdb) current state.

So we would need to generate MPT, storage, but then what about serializers, deserializers, MPT can have different types for keys and values (usually those are `Array[Byte]` but for storage (smart contract code) it is `BigInt`.

But there are already property tests. I changed this guy using the approach. (I needed to understand those tests and I noticed some duplications - so there are so helper methods extracted).

WDYT is it good enough?

Reply:
Looks good yes, I wasn't aware about the MPT being coupled with the storage, but I'll have to learn more about that :)